dap_prinz_green_jobs/analysis/ojo_analysis/process_ojo_green_measures.py

```diff
@@ -0,0 +1,753 @@
+    soc_descriptions_dict=None,
+    agg_col="SOC_2020_EXT",
+    job_id_col="job_id",
+):
+    """
+    Much like create_agg_occ_measures but more generic to aggregate by any column
+    """
+    aggregated_data = {}
+    for agg_value in tqdm(all_green_measures_df[agg_col].unique()):
+        if pd.notnull(agg_value):
+            filtered_data = all_green_measures_df[
+                all_green_measures_df[agg_col] == agg_value
+            ]
+
+            filtered_skills = green_skills_df[
+                green_skills_df[job_id_col].isin(
+                    set(filtered_data[job_id_col].tolist())
+                )
+            ]
+
+            top_green_skill_num = (
+                filtered_skills[
+                    ["green_skill_preferred_name", "extracted_green_skill_id"]
+                ]
+                .value_counts()[0:5]
+                .to_dict()
+            )
+            top_green_skill_prop = (
+                filtered_skills[
+                    ["green_skill_preferred_name", "extracted_green_skill_id"]
+                ]
+                .value_counts(normalize=True)[0:5]
+                .to_dict()
+            )
+
+            green_skill_info = []
+            for k, v in top_green_skill_num.items():
+                green_skill_info.append(
+                    {
+                        "skill_name": k[0],
+                        "skill_id": k[1],
+                        "num_job_ads": v,
+                        "prop_job_ads": top_green_skill_prop[k],
+                    }
+                )
+
+            top_5_not_green_skills_num = (
+                filtered_skills[pd.isnull(filtered_skills["extracted_green_skill"])][
+                    ["full_skill_preferred_name", "extracted_full_skill_id"]
+                ]
+                .value_counts()[0:5]
+                .to_dict()
+            )
+            top_5_not_green_skills_prop = (
+                filtered_skills[pd.isnull(filtered_skills["extracted_green_skill"])][
+                    ["full_skill_preferred_name", "extracted_full_skill_id"]
+                ]
+                .value_counts(normalize=True)[0:5]
+                .to_dict()
+            )
+
+            not_green_skill_info = []
+            for k, v in top_5_not_green_skills_num.items():
+                not_green_skill_info.append(
+                    {
+                        "skill_name": k[0],
+                        "skill_id": k[1],
+                        "num_job_ads": v,
+                        "prop_job_ads": top_5_not_green_skills_prop[k],
+                    }
+                )
+
+            top_5_sics_num = (
+                filtered_data[["SIC_2_digit", "SIC_2_digit_name"]]
+                .value_counts()[0:5]
+                .to_dict()
+            )
+            top_5_sics_prop = (
+                filtered_data[["SIC_2_digit", "SIC_2_digit_name"]]
+                .value_counts(normalize=True)[0:5]
+                .to_dict()
+            )
```

Comment:
I can't decide if 2 digit is the way to go in the end... because it kind of obfuscates things - something to revisit!